### ISSUE TITLE
runworker doesn't react to Ctrl-C, only terminates on SIGKILL

### ISSUE THREAD

**Issue by mikaraunio** [proofit404/asgi_rabbitmq#15]
Moved from django/channels#764

- Your OS and runtime environment, and browser if applicable

Debian GNU/Linux 9.1 (stretch)
Linux hostname 4.9.0-3-amd64 #1 SMP Debian 4.9.30-2+deb9u3 (2017-08-06) x86_64 GNU/Linux
python2.7 2.7.13-2
rabbitmq-server 3.6.6-1

- The versions of Channels, Daphne, Django, Twisted, and your ASGI backend (asgi_ipc or asgi_redis normally)

asgi-ipc==1.4.1                                                                                     
asgi-rabbitmq==0.5.3
asgiref==1.1.2
daphne==1.3.0
Django==1.11.5
pika==0.10.0
Twisted==17.9.0

- What you expected to happen vs. what actually happened

When using the IPC channel layer, runworker exits when Ctrl-C is pressed, as expected. When using the RabbitMQ channel layer, Ctrl-C has no effect on runworker. Ctrl-Z + `kill -9` is required to stop the worker.

- How you're running Channels (runserver? daphne/runworker? Nginx/Apache in front?)

Nginx + uwsgi + daphne + runworker

- Console logs and full tracebacks of any errors
```
$ manage.py runworker -v2
2017-10-12 16:22:14,033 - INFO - runworker - Using single-threaded worker.
2017-10-12 16:22:14,034 - INFO - runworker - Running worker against channel layer default (asgi_rabbitmq.core.RabbitmqChannelLayer)
2017-10-12 16:22:14,035 - INFO - worker - Listening on channels http.request, websocket.connect, websocket.disconnect, websocket.receive
^C^C^C^C^C^C

^Z
[1]+  Stopped
$ kill %1
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "build/bdist.linux-x86_64/egg/asgi_rabbitmq/core.py", line 735, in run
    self.connection.run()
  File "build/bdist.linux-x86_64/egg/asgi_rabbitmq/core.py", line 627, in run
    self.connection.ioloop.start()
  File "build/bdist.linux-x86_64/egg/pika/adapters/select_connection.py", line 354, in start
    self.poll()
  File "build/bdist.linux-x86_64/egg/pika/adapters/select_connection.py", line 590, in poll
    events = self._poll.poll(self.get_next_deadline())
IOError: [Errno 4] Interrupted system call
$ kill %1
$ kill -9 %1
[1]+  Killed
$
```

**Comment by proofit404** [comment]
Hi, thanks for using my project!

Unfortunately, I can't reproduce reported issue.
I tried following things:
1. Run it with `./manage.py runserver`
2. Run it with `daphne` and `runworker` separately.
3. Run it with `./manage.py runserver` inside docker
4. Run it with `daphne` and `runworker` separately inside docker.
5. Boot python shell and run channels by invoking code manually.
6. Boot python shell and run channels by invoking code manually inside docker.

In each step it tried to stop processes under load and in the idle state. Each time it ends up with expected teardown behavior.

I can't pospond new release so I'm going to mark it us can't reproduce.

Please, provide more complete example with deterministic reproducing steps.

Regards, Artem.

**Comment by mikaraunio** [comment]
Hi Artem,

Thanks for looking into this.

I will try and bundle a minimal test project for reproducing this next week. I can confirm that for me, the issue happens even without daphne or any other components running, just by invoking `./manage.py runworker` from the shell and then trying to Ctrl-C it.

I upgraded to asgi-rabbitmq 0.5.4 and pika 0.11, this made no difference.

Having a quick look at the code, I noticed that pika has a couple of bare `except:` clauses that would consume KeyboardInterrupt exceptions. However, replacing those with `except Exception:` didn't help either.
